### PR TITLE
User story 20

### LINKED ISSUE #14: User Story 20, User can delete an author
```
As a Visitor,
When I visit an author's show page,
I see a link on the page to delete the author.
This link should return me to the book index page where I
no longer see this author listed.
If this author was the only author for any book, that book is also deleted.
If this author co-authored a book with someone else, that book should also be deleted, but not the other author.

(you may need to delete other content before you can delete an author or book)
```

DO THESE STORIES FIRST:
- User Story 14, Author Show Page

### Checklist:

- [x] base code tests are written
- [x] base code is written to pass tests
- [x] edge case ("sad path") tests written
- [x] edge case code written
- [x] code is reviewed
- [x] branch is merged
- [x] all tests still pass

child of #9

child of #33

depends on #14

### PR DESCRIPTION
**Pull request by chakeresa**
This PR/branch...
 - Resolves #14 (User story 20: destroy author)
 - Adds a button to the author show page to destroy that author
 - Destroys the author, their books, and the reviews of their books
 - Redirects to the books index
 - Tests everything above

### CODE REVIEW

File: app/controllers/authors_controller.rb [hunk 1]
```diff
@@ -9,4 +9,17 @@ def show
 
   def index
   end
+
+  def destroy
+    relevant_book_authors_ids = BookAuthor.where(author_id: params[:id].to_i).pluck(:id)
```
Comment (DavisC0801) [inline app/controllers/authors_controller.rb]: For MVC we should be doing the Where and pluck call in the model.

**Review by DavisC0801 — COMMENTED** [review]
Rewrite most of the controller action to change database queries to model work.

File: app/controllers/authors_controller.rb (hunk 1 above)
Comment (chakeresa) [reply app/controllers/authors_controller.rb]: Done - moved!

**Review by DavisC0801 — APPROVED** [review]
Refactor follows MVC, no further refactors needed.
